### PR TITLE
Configurable tag process

### PR DESCRIPTION
**Pull request by newhoggy**
The `cardano-dev.yaml` file contains options to change things like main branch and release branches

### CODE REVIEW

File: scripts/tag.sh [hunk 1]
```diff
@@ -16,7 +16,10 @@
 
 set -euo pipefail
 
-main_branch="main"
+git_root="$(git rev-parse --show-toplevel)"
+main_branch="$(cat "$git_root/cardano-dev.yaml" | yq -o json | jq -r '."main-branch"')"
```
Comment (carbolymer) [inline scripts/tag.sh]: This would require cardano-dev.yaml file. Can you link to an example from here, or provide it in the repo?

**Review by carbolymer — APPROVED** [review]

File: scripts/tag.sh (hunk 1 above)
Comment (newhoggy) [reply scripts/tag.sh]:
```
cat ../cardano-api/cardano-dev.yaml
version:
main-branch: main
release-branch:
  prefix: release/
  suffix: -branch
```

Comment (newhoggy) [reply scripts/tag.sh]: I'm going to convert this to a dotfile.

Comment (carbolymer) [reply scripts/tag.sh]: By 'here' I meant the comment in `tag.sh` file. 😃

**Pull request merged**